flink-clients/src/main/java/org/apache/flink/client/cli/ExecutionConfigAccessor.java

```diff
@@ -74,6 +74,11 @@ public static ExecutionConfigAccessor fromProgramOptions(final ProgramOptions op
 
 		SavepointRestoreSettings.toConfiguration(options.getSavepointRestoreSettings(), configuration);
 
+		// If the Python configuration exists, write them to the configuration for Python program/UDF execution.
+		if (options instanceof PythonProgramOptions) {
```

Comment:
Instead of the `instanceof` here, I think it's better to add a method to `ProgramOptions` for adding additional configuration. Something like `(addToConfiguration(Configuration)`. On the base `ProgramOptions` this would do nothing but in `PythonProgramOptions` this would add the additional python configuration.

If we really want to do it nicely, though, we can even add the current code in this method here to the new `ProgramOptions.addToConfiguration()`. This way we treat everything the same.

Reply:
Ah, better name is probably `applyToConfiguration()`, and hand in a `Configuration`.